osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/config/AbstractSchedulingConfig.java

```diff
@@ -114,15 +124,81 @@ protected Scheduler constructScheduler(final SchedulingConfigProperties scheduli
     protected Scheduler constructAndStartQuartzScheduler(final SchedulingConfigProperties schedulingConfigProperties)
             throws SchedulerException {
 
+        Scheduler scheduler = null;
+        try {
+            scheduler = this.constructSchedulerInstance(schedulingConfigProperties);
+        } catch (final Exception e) {
+            LOGGER.error("Exception", e);
+            scheduler = this.retryConstructAndStartQuartzScheduler(schedulingConfigProperties);
+        }
+
+        scheduler.start();
+
+        return scheduler;
+    }
+
+    private Scheduler retryConstructAndStartQuartzScheduler(final SchedulingConfigProperties schedulingConfigProperties)
+            throws SchedulerException {
+
+        this.startAttemptCount++;
+
+        if (this.startAttemptCount < this.maxStartAttemptCount) {
+            try {
+                try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1141)